### ISSUE TITLE
builder/virtualbox: optionally skip uploading VBoxVersionFile

### ISSUE THREAD

**Issue by lamoni** [mitchellh/packer#3619]
cannot use `none` communicator in virtualbox builder because it tries to upload the `VBoxVersionFile` with no way to disable it.

Separating this into it's own issue.  This is related to #3618 (it's actually the cause of that issue).

Disabling the vbox version file upload isn't working due to conflicting code in vbox_version_config.go and step_upload_version.go:

vbox_version_config.go:

```
func (c *VBoxVersionConfig) Prepare(ctx *interpolate.Context) []error {
    if c.VBoxVersionFile == "" {
        c.VBoxVersionFile = ".vbox_version"
    }

    return nil
}
```

step_upload_version.go

```
func (s *StepUploadVersion) Run(state multistep.StateBag) multistep.StepAction {
    comm := state.Get("communicator").(packer.Communicator)
    driver := state.Get("driver").(Driver)
    ui := state.Get("ui").(packer.Ui)

    if s.Path == "" {
        log.Println("VBoxVersionFile is empty. Not uploading.")
        return multistep.ActionContinue
    }
```

In short, vbox_version_config.go makes it impossible to set an empty string as the value, while the only way to disable version file uploading is to set it to an empty string.

Fix incoming.

**Comment by lamoni** [comment]
Is it desired to have the default value?

I ask because the step_upload_version_test.go needs to be fixed if so, as it's not currently instantiating test config.

The test passes despite the code this is testing not actually working as expected:

```
func TestStepUploadVersion_noPath(t *testing.T) {
	state := testState(t)
	step := new(StepUploadVersion)
	step.Path = ""

	comm := new(packer.MockCommunicator)
	state.Put("communicator", comm)

	// Test the run
	if action := step.Run(state); action != multistep.ActionContinue {
		t.Fatalf("bad action: %#v", action)
	}
	if _, ok := state.GetOk("error"); ok {
		t.Fatal("should NOT have error")
	}

	// Verify
	if comm.UploadCalled {
		t.Fatal("bad")
	}
}
```

**Comment by rickard-von-essen** [comment]
Yes we should keep the default `.vbox_version`.
